Casks/d/data-integration.rb

```diff
@@ -6,11 +6,11 @@
       verified: "privatefilesbucket-community-edition.s3.amazonaws.com/"
   name "Pentaho Data Integration"
   desc "End to end data integration and analytics platform"
-  homepage "https://www.hitachivantara.com/en-us/products/pentaho-platform/data-integration-analytics/pentaho-community-edition.html"
+  homepage "https://pentaho.com/pentaho-developer-edition/"
 
   livecheck do
-    url :homepage
-    regex(/href=.*?pdi-ce[._-]v?(\d+(?:[.-]\d+)+)\.zip/i)
+    url "https://privatefilesbucket-community-edition.s3.amazonaws.com/"
+    regex(%r{/ce/client-tools/psw-ce-(\d+(?:[.-]\d+)+)(?!.*/ce/client-tools/psw-ce-(\d+(?:[.-]\d+)+))}i)
```

Comment:
```suggestion
    regex(/pdi-ce[._-]v?(\d+(?:[.-]\d+)+)\.zip/i)
    strategy :xml do |xml, regex|
      xml.get_elements("//Contents/Key").map do |item|
        match = item.text&.strip&.match(regex)
        next if match.blank?

        match[1]
      end
    end
```

We can continue using the previous `regex` and omit the `href` attribute since the filename has not changed. To reduce the search space, we can parse the XML document and only match the text in the `key` elements.